appservice/src/deploy/getDeployFsPath.ts

```diff
@@ -19,7 +19,7 @@ const deploySubpathSetting: string = 'deploySubpath';
  * In App Service, users can deploy specific artifact files (such as .jar) which is handled by selectWorkspaceFile
  * This enforces that the `effectiveDeployFsPath` is currently open in a workspace
  */
-export async function getDeployFsPath(context: IActionContext, target: vscode.Uri | string | AzExtTreeItem | undefined, fileExtensions?: string | string[]): Promise<IDeployPaths> {
+export async function getDeployFsPath(context: IActionContext, target: vscode.Uri | string | AzExtTreeItem | undefined, fileExtensions?: string | string[], recommendations?: IAzureQuickPickItem<string>[]): Promise<IDeployPaths> {
```

Comment:
It's strange to me that `recommendations` is a `IAzureQuickPickItem[]`. In general, I'd expect that this would accept a `string[]` with the fspath of the recommendations and that it should be converted into quickPickItems by selectWorkspaceFile.

Reply:
@nturinski  

**I updated to use a "better" word `suggestion` instead of `recommendation`.**

first, `string` is not enough to meet my requirements, because the `label` & `data` & `description` of the final `IAzureQuickPickItem` are different in my case.
e.g. 
![image](https://user-images.githubusercontent.com/69189193/118904947-70bd3300-b94d-11eb-9278-c2c742d39ef2.png)

the view model for the first suggested item is actually:
```
{
  label: 'Maven: com.example:spring-boot:0.0.1-SNAPSHOT:jar'
  description: 'initial\complete\spring-boot-0.0.1-SNAPSHOT:jar'
  data: `$workspace\gs-spring-boot\complete`
}
```